common/config/config.go

```diff
@@ -41,37 +41,44 @@
 }
 
 func GetConfig(cmd *cobra.Command) (*Config, error) {
-	if !cmd.Flags().Changed(flagForConfigFile) {
-		// The config flag was not set. Attempt to find the config via environment variables or in the local directory
-		return loadConfig("")
-	}
-	// The config flag was explicitly set
-	configFile, err := cmd.Flags().GetString(flagForConfigFile)
+	cfg, err := findAndLoadConfigFile(cmd)
 	if err != nil {
 		return nil, err
 	}
-	if configFile == "" {
-		return nil, errors.New("config cannot be empty")
-	}
-	return loadConfig(configFile)
+
+	// Set any default values.
+	cfg.Build = true
+	return cfg, nil
 }
 
-func loadConfig(filename string) (*Config, error) {
-	if filename != "" {
-		return loadConfigFromFile(filename)
+// findAndLoadConfigFile searches for a config file based on the following priorities:
+// 1. A config file set via a flag
+// 2. A config file set via an environment variable
+// 3. A config file named "world.toml" in the current directory
+// 4. A config file found in a parent directory.
+func findAndLoadConfigFile(cmd *cobra.Command) (*Config, error) {
+	// First look for the config file in the config file flag.
+	if cmd.Flags().Changed(flagForConfigFile) {
```

Comment:
I try to run `world cardinal start --config=/home/daim/starter-game/world.toml` command but got error `Error: unknown flag: --config`

After i'm checking it --config flag is only added to rootCmd for now.

Need to add `--config` flag for `startCmd, devCmd, restartCmd, purgeCmd, stopCmd`

Reply:
Good catch. I changed the config flag to a "PersistentFlag" which sets it for all sub-commands as well.